conbench/app/_plots.py

```diff
@@ -152,7 +152,7 @@ def simple_bar_plot(benchmarks, height=400, width=400, vbar_width=0.7):
 
 # Using `Optional[float]` here only because of BenchmarkResult.mean being
 # nullable as of today.
-def _should_format(floats: list[Optional[float]], unit):
+def _should_format(floats: List[Optional[float]], unit):
```

Comment:
I took the opportunity to revert to the backwards-compatible typing containers here, just in case (see https://github.com/conbench/conbench/pull/811#discussion_r1131556101). Agreed that we should put a framework in place for ensuring we never accidentally break the legacy `conbench` CLI running in the arrow-benchmarks-ci repo.